File: Baekjoon/강의/강의 기초편/Brute Force/순열/10819.차이를최대로.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
int cal(vector<int>& a) {
	int result = 0;
	for (int i = 1; i < a.size(); i++) {
		result += abs(a[i - 1] - a[i]);
	}
	return result;
}
int main() {
	int n;
	int ans=0;
	cin >> n;
	vector<int> a(n);
	for (int i = 0; i < n; i++) {
		cin >> a[i];
	}
	sort(a.begin(), a.end());
	do {
		int temp = cal(a);
		ans = max(ans,temp);
	} while (next_permutation(a.begin(), a.end()));
	cout << ans << '\n';
	return 0;
}

File: Baekjoon/강의/강의 기초편/Brute Force/순열/10819.차이를최대로.py
from itertools import permutations

n=int(input())
a=list(map(int,input().split()))
ans=0

for b in permutations(a,n):
    tmp=0
    for i in range(n-1):
        tmp+=abs(b[i]-b[i+1])
    ans=max(ans,tmp)

print(ans)